File: src/modules/regex.ts
// Any character Discord uses for Markdown.
export const markdown = /[*_`~]/;

// Any character that can be the beginning of a special string in Discord.
export const discordStringBeginning = /[<@:]/;

// Things that should not be capitalized:
// - emojis (start with < and end with >)
// - URLs (contain '://' somewhere in the middle)
export const doNotCapitalize = /(^<.*>$)|(^.+:\/\/.+$)/;

File: utils/regex.py
import re

markdown = re.compile(r"[*_`~]")
discord_string_start = re.compile(r"[<@:]")
do_not_capitalize = re.compile(r"(^<.*>$)|(^.+:\/\/.+$)")
url = re.compile(r"^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$")
